xonsh/completer.py

```diff
@@ -78,6 +78,21 @@ def _normpath(p):
 
     return p
 
+def alias_expand(line):
+    """ Expands any aliases present in line if alias does not point to a builtin function and if alias is only a single command.
+    """
+    
+    for word in line.split(' '):
+        if (word in builtins.aliases
+            and type(builtins.aliases.get(word)) is list
+            and len(builtins.aliases.get(word)) == 1):
+            word_idx = line.find(word)
+            line = line[:word_idx] +\
+                    builtins.aliases.get(word)[0] +\
+                    line[word_idx+len(word):]
```

Comment:
The reason for this is that you could have aliases defined in xonsh that end up not being valid because they are not defined elsewhere (like in git).